nuxeo-core/nuxeo-core-test/src/test/java/org/nuxeo/ecm/core/event/test/TransitionTest.java

```diff
@@ -54,10 +59,34 @@
 
     private String lastComment;
 
+    protected EventListenerDescriptor eventDesc;
+
     @Before
     public void doBefore() {
         lastComment = null;
-        registerAsEventListenerFor(org.nuxeo.ecm.core.api.LifeCycleConstants.TRANSITION_EVENT);
+        final EventListener el = this;
+        eventDesc = new EventListenerDescriptor() {
+            @Override
+            public Set<String> getEvents() {
+                return Sets.newHashSet(LifeCycleConstants.TRANSITION_EVENT);
+            }
+
+            @Override
+            public EventListener asEventListener() {
+                return el;
```

Comment:
You don't need a local variable `el`, you can just do `return TransitionTest.this`